### PR TITLE
Fix copyFrom sub-class object

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-6179

## Issue
- PropertyInfo.isSet() is checking `<propName>IsSet_` private local variable on the class itself, when copying from a sub-class object which has it own `<propName>IsSet_` was not being checked.

## Changes
- Check if the obj to copy is the same class as `this` and act accordingly

### CODE REVIEW

**Review by jlhughes — COMMENTED** [review]
This is similar to the ordering of overlay, which first gets properties through the 'obj' argument, then falls back to 'this' properties. 
Perhaps copyFrom should do the same.

**Review by nanoscott — APPROVED** [review]

**Review by jlhughes — APPROVED** [review]

File: src/foam/core/FObject.java [hunk 1]
```diff
@@ -299,16 +299,16 @@ default FObject copyFrom(FObject obj) {
     List<PropertyInfo> props = getClassInfo().getAxiomsByClass(PropertyInfo.class);
     for ( PropertyInfo p : props ) {
       try {
-        if ( p.isSet(obj) ) p.set(this, p.get(obj));
-      } catch (ClassCastException e) {
-        try {
+        if ( getClass().equals(obj.getClass()) ) {
```
Comment (kgrgreer) [inline src/foam/core/FObject.java]: Use == instead. @penzital This is caused by the same instance variable overriding issue that you detected. If we merge this, we should create a ticket to fix the underlying issue and then revert this change.

Comment (lchanmann) [reply src/foam/core/FObject.java]: Fixed in https://github.com/kgrgreer/foam3/pull/1264/commits/a8836971de3d56bf006bac4900c8729b9a1cf40e.
